### ISSUE TITLE
inotifywait 3.14 crashes on Ubuntu 17.10 with certain directory names

### ISSUE THREAD

**Issue by mulle-nat** [inotify-tools/inotify-tools#84]
Terminal A:

```
$ inotifywait -r -m -c /tmp
Setting up watches.  Beware: since -r was given, this may take a while!
Watches established.
```

Terminal B:

```
$ mkdir /tmp/v,f
$ touch /tmp/v,f/l,bo
```

Terminal A:

```
/tmp/,"CREATE,ISDIR","v,f"
/tmp/,"OPEN,ISDIR","v,f"
/tmp/,"ACCESS,ISDIR","v,f"
/tmp/,"CLOSE_NOWRITE,CLOSE,ISDIR","v,f"
Segmentation fault (core dumped)
```

On ubuntu 17.10: Linux 4.13.0-25-generic #29-Ubuntu SMP Mon Jan 8 21:14:41 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

**Comment by 813gan** [comment]
I need to make more tests, but it looks like this bug disappears in https://github.com/813gan/inotify-tools/tree/NULL-in-format-string

**Comment by mulle-nat** [comment]
For Ubuntu/debian there is now a slightly patched version of inotify-tools (3.14-6) that doesn't exhibit the bug, whereas (3.14-2) is buggy.  There is no easy way to differentiate both, which is a bit unfortunate.
